fractal_server/app/runner/__init__.py

```diff
@@ -249,15 +249,18 @@ async def submit_workflow(
         # Replace output_dataset.meta with output_dataset_meta, while handling
         # the history property in a special way (i.e. appending to and
         # existing entry rather than replacing it)
-        new_meta = {}
-        for key, value in output_dataset_meta.items():
-            if key != "history":
-                # For non-history keys, replace with new value
-                new_meta[key] = value
-            else:
-                # For history key, append to existing entry
-                new_meta[key] = output_dataset.meta.get(key, []) + value
-        output_dataset.meta = new_meta
+        output_dataset.history = output_dataset_meta.pop("history")
```

Comment:
This doesn't sound fully correct, at least in terms of variable name.s

What is the output of `process_workflow`?
1. If it's a instance of `meta`, then we can keep calling it `output_dataset_meta`. But then it should not include a `history` key.
2. If it's a collection of both the new meta and history, then let's make it something like `output_whatever=dict(meta=..., history=...).
3. Another option is that `process_workflow` returns a tuple, with meta and history.

I'd be in favor of option 2, but all of them are OK.